Source/diablo.cpp

```diff
@@ -2181,4 +1940,275 @@ void diablo_color_cyc_logic()
 	}
 }
 
+void helpKeyPressed()
+{
+	if (helpflag) {
+		helpflag = false;
+	} else if (stextflag != STORE_NONE) {
+		ClearPanel();
+		AddPanelString(_("No help available"), true); /// BUGFIX: message isn't displayed
+		AddPanelString(_("while in stores"), true);
+		track_repeat_walk(false);
+	} else {
+		invflag = false;
+		chrflag = false;
+		sbookflag = false;
+		spselflag = false;
+		if (qtextflag && leveltype == DTYPE_TOWN) {
+			qtextflag = false;
+			stream_stop();
+		}
+		questlog = false;
+		automapflag = false;
+		msgdelay = 0;
+		gamemenu_off();
+		DisplayHelp();
+		doom_close();
+	}
+}
+
+void itemInfoKeyPressed()
+{
+	if (pcursitem != -1) {
+		sprintf(
+		    tempstr,
+		    "IDX = %i  :  Seed = %i  :  CF = %i",
+		    items[pcursitem].IDidx,
+		    items[pcursitem]._iSeed,
+		    items[pcursitem]._iCreateInfo);
+		NetSendCmdString(1 << myplr, tempstr);
+	}
+	sprintf(tempstr, "Numitems : %i", numitems);
+	NetSendCmdString(1 << myplr, tempstr);
+}
+
+void inventoryKeyPressed()
+{
+	if (stextflag == STORE_NONE) {

```

Comment:
Would prefer the use of early return for things like this.

Reply:
I just used the original code. :)
Sure, I will do some cleanup.

Reply:
It would be helpful, but not required for getting this merged :)